tests/jest/views/Dashboard.spec.js

```diff
@@ -0,0 +1,194 @@
+/* jshint esversion: 8 */
+
+import { shallowMount, mount, createLocalVue } from '@vue/test-utils'
+import Dashboard from '../../../src/views/Dashboard'
+import axios from '@nextcloud/axios'
+import * as initialState from '@nextcloud/initial-state'
+import { STATE } from '../../../src/utils'
+import notificationsResponse from '../fixtures/notificationsResponse.json'
+import * as dialogs from '@nextcloud/dialogs'
+
+jest.mock('@nextcloud/axios')
+jest.mock('@nextcloud/l10n', () => ({
+	translate: jest.fn((app, msg) => msg),
+	getLanguage: jest.fn(),
+}))
+jest.mock('@nextcloud/dialogs', () => ({
+	getLanguage: jest.fn(() => ''),
+	showError: jest.fn(),
+	showSuccess: jest.fn(),
+}))
+
+global.OCA = {}
+global.OC = {}
+const localVue = createLocalVue()
+
+describe('Dashboard.vue', () => {
+	const dashboardTriggerButtonSelector = '.trigger'
+	const markAsReadButtonSelector = '.popover__wrapper .action-button'
+	let wrapper
+	beforeEach(() => {
+		// eslint-disable-next-line no-import-assign
+		initialState.loadState = jest.fn(() => true)
+
+		// mock the beforeMount() method, so that the loop is not called automatically
+		// we first need to mount the component, and the tests will call the loop themselves
+		Dashboard.beforeMount = jest.fn()
+	})
+	it('should show the notification items in the Dashboard', async () => {
+		wrapper = shallowMount(
+			Dashboard,
+			{
+				localVue,
+				mocks: {
+					t: (app, msg) => msg,
+					generateUrl() {
+						return '/'
+					},
+				},
+				propsData: {
+					title: 'dashboard',
+				},
+			})
+		const axiosSpy = jest.spyOn(axios, 'get')
+			.mockImplementationOnce(() => Promise.resolve({ data: 'http://openproject.org' }))
+			.mockImplementationOnce(() => Promise.resolve({ data: notificationsResponse }))
+		await wrapper.vm.launchLoop()
+		expect(axiosSpy).toBeCalledWith(
+			'http://localhost/apps/integration_openproject/url',
+		)
+		expect(axiosSpy).toBeCalledWith(
+			'http://localhost/apps/integration_openproject/notifications',
+		)
+		expect(wrapper.vm.state).toBe(STATE.OK)
+		expect(wrapper.vm.items).toStrictEqual([
```

Comment:
maybe saving as the snapshot is better. that way long expected array is not necessary. also easy to maintain.
```suggestion
		expect(wrapper.vm.items).toMatchSnapshot()
```

Reply:
done! I was not aware that we can use snapshot on a arbitrary object, but that totally makes sense